client/src/components/Overview/Overview.jsx

```diff
@@ -10,67 +10,62 @@ class Overview extends React.Component {
     super(props)
     this.state = {
       currentProduct: null,
-      productStyles: [],
-      currentStyle: null
+      productStyles: null,
+      currentStyle: null,
+      onSale: false
     }
-    this.getProductStyle = this.getProductStyle.bind(this);
-    this.getProductDetails = this.getProductDetails.bind(this);
+    this.getDefaultStyle = this.getDefaultStyle.bind(this);
   }
 
   componentDidUpdate(oldProps) {
     if(this.props.product !== oldProps.product) {
-      // this.setState({currentProduct: this.props.product})
-      this.getProductStyle()
-      this.getProductDetails()
+      this.setState({currentProduct: this.props.product});
+    }
+    if (this.props.styles !== oldProps.styles) {
+      this.setState({productStyles: this.props.styles});
     }
   }
 
-  getProductStyle() {
-    $.get({
-      url: `https://app-hrsei-api.herokuapp.com/api/fec2/hr-rfe/products/${this.props.product.id}/styles`,
-      headers: {Authorization: Token},
-      success: (data) => {
-        this.setState({productStyles: data})
-      },
-      error: (err) => {
-        console.log(err)
+  getDefaultStyle(styles) {
+    styles.forEach((style) => {
+      if (style['default?'] === true) {
+        this.setState({currentStyle: style})
       }
     })
   }
 
-  getProductDetails() {
-    $.get({
-      url: `https://app-hrsei-api.herokuapp.com/api/fec2/hr-rfe/products/${this.props.product.id}`,
-      headers: {Authorization: Token},
-      success: (data) => {
-        this.setState({currentProduct: data})
-      },
-      error: (err) => {
-        console.log(err)
-      }
-    })
-  }
+
 
 
 
   render() {
     if (!this.state.currentProduct) {
```

Comment:
You have to check for both product and styles being null, since they upload at different times

if (!this.state.currentProduct || !this.state.productStyles)